dot/core/service.go

```diff
@@ -512,8 +515,12 @@ func (s *Service) createBlockResponse(msg *network.BlockRequestMessage) (*networ
 
 	switch c := msg.StartingBlock.Value().(type) {
 	case uint64:
+		if c == 0 {
+			c = 1
+		}
+
 		block, err := s.blockState.GetBlockByNumber(big.NewInt(0).SetUint64(c))
-		if err != nil {
+		if err != nil || block == nil {
```

Comment:
do you think that s.blockState.GetBlockByNumber returning a nil block and no error is a acceptable logic for the method ? shall we be doing double checks like this or update the logic in that method to avoid having to double check it ?

Reply:
yeah you're right I don't think it will ever return nil, will remove the `block == nil` part